monai/losses/shape.py

```diff
@@ -0,0 +1,172 @@
+        smooth_nr: float = 1e-8,
+        smooth_k: float = 2e-1,
+        reduction: Union[LossReduction, str] = LossReduction.MEAN,
+    ) -> None:
+        """
+        Shape Information Loss inspired by [Huang et al., 2021](https://ieeexplore.ieee.org/document/9433775)
+
+        Args:
+            include_background: if False, channel index 0 (background category) is excluded from the calculation.
+                if the non-background segmentations are small compared to the total image size they can get overwhelmed
+                by the signal from the background so excluding it in such cases helps convergence.
+            to_onehot_y: whether to convert `y` into the one-hot format. Defaults to False.
+            sigmoid: if True, apply a sigmoid function to the prediction.
+            softmax: if True, apply a softmax function to the prediction.
+            other_act: if don't want to use `sigmoid` or `softmax`, use other callable function to execute
+                other activation layers, Defaults to ``None``. for example:
+                `other_act = torch.tanh`.
+            smooth_nr: a small constant added to the numerator to avoid zero.
+            smooth_k: smoothness factor used in the Heaviside function
+            reduction: {``"none"``, ``"mean"``, ``"sum"``}
+                Specifies the reduction to apply to the output. Defaults to ``"mean"``.
+
+                - ``"none"``: no reduction will be applied.
+                - ``"mean"``: the sum of the output will be divided by the number of elements in the output.
+                - ``"sum"``: the output will be summed.
+
+        Raises:
+            TypeError: When ``other_act`` is not an ``Optional[Callable]``.
+            ValueError: When more than 1 of [``sigmoid=True``, ``softmax=True``, ``other_act is not None``].
+                Incompatible values.
+        """
+        super().__init__(reduction=LossReduction(reduction).value)
+        if other_act is not None and not callable(other_act):
+            raise TypeError(f"other_act must be None or callable but is {type(other_act).__name__}.")
+        if int(sigmoid) + int(softmax) + int(other_act is not None) > 1:
+            raise ValueError("Incompatible values: more than 1 of [sigmoid=True, softmax=True, other_act is not None].")
+        self.include_background = include_background
+        self.to_onehot_y = to_onehot_y
+        self.sigmoid = sigmoid
+        self.softmax = softmax
+        self.other_act = other_act
+        self.smooth_nr = float(smooth_nr)
+        self.smooth_k = float(smooth_k)
+
+    def distance_map(self, mask: torch.Tensor) -> torch.Tensor:
+        """Creates a distance map from a 2D binary mask.
+
+        Args:
+            mask (torch.Tensor): Binary mask of shape [WD].
+
+        Returns:
+            torch.Tensor: Boundary distance map of shape [WD]
+        """
+        # Convert to NumPy to use with SciPy
+        roi: np.ndarray = mask.numpy()
```

Comment:
please use `monai.utils.convert_data_type`: `roi = convert_data_type(mask, np.ndarray)`, it can handle the device properly...